Fix systemd_dep assertions to not fail when passing a valid service directory path.

diff --git a/data/meson.build b/data/meson.build
index 2cfe4691..2211e98b 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -166,20 +166,20 @@ endif
 
 systemdsystemunitdir = get_option('systemdsystemunitdir')
 if systemdsystemunitdir != 'no'
-  assert(systemd_dep.found(), 'systemd required but not found, please provide a valid systemd system unit dir or disable it')
   if get_option('systemdsystemunitdir') != ''
     systemd_systemunitdir = get_option('systemdsystemunitdir')
   else
+    assert(systemd_dep.found(), 'systemd required but not found, please provide a valid systemd system unit dir or disable it')
     systemd_systemunitdir = systemd_dep.get_pkgconfig_variable('systemdsystemunitdir')
   endif
 endif
 
 systemduserunitdir = get_option('systemduserunitdir')
 if systemduserunitdir != 'no'
-  assert(systemd_dep.found(), 'systemd required but not found, please provide a valid systemd user unit dir or disable it')
   if get_option('systemduserunitdir') != ''
     systemd_userunitdir = get_option('systemduserunitdir')
   else
+    assert(systemd_dep.found(), 'systemd required but not found, please provide a valid systemd user unit dir or disable it')
     systemd_userunitdir = systemd_dep.get_pkgconfig_variable('systemduserunitdir',
                                                              define_variable: ['prefix', get_option('prefix')])
   endif
